serialize.js

```diff
@@ -0,0 +1,33 @@
+export function serialize(node) {
+  if (node === undefined || node === null) {
+    return "null";
+  }
+
+  const { value, left, right } = node;
+
+  return `${value},${serialize(left)},${serialize(right)}`;
+}
+
+export function deserialize(str) {
+  const nodes = str.split(",");
+
+  function buildTree() {
+    if (nodes.length === 0) {
+      return null;
```

Comment:
In theory, this should never happen since you're not supposed to call `buildTree` more than the count of nodes. But... this might actually happen in case the tree was serialized wrong (or the data was lost during transmission). That's actually a good use case! let me add a test for that, and you should throw an error here `throw new Error('bad format')`